help/_plugins/SitePostRender.rb

```diff
@@ -21,50 +21,255 @@ def self.process_page(page)
+        "import type {ReactNode} from 'react';",
+        "import React from 'react';",
+        "import {View} from 'react-native';",
+      ]
 
-        # Construct the full hierarchical ID by concatenating IDs for all levels up to the current level
-        full_id = (2..level).map { |l| prefix[l] }.join('--')
+      # Add conditional imports based on component usage
+      imports << "import BulletList from '@components/SidePanel/HelpComponents/HelpBulletList';" if help_content_string.include?("<BulletList")
+      imports << "import Text from '@components/Text';" if help_content_string.include?("<Text")
+      imports << "import TextLink from '@components/TextLink';" if help_content_string.include?("<TextLink")
+      imports << "import type {ThemeStyles} from '@styles/index';"
 
-        # Assign the generated ID to the header element
-        header['id'] = full_id
+      # Join the imports
+      import_block = imports.join("\n")
 
-        puts "    Found h#{level}: '#{header_text}' -> ID: '#{full_id}'"
+      ts_output = <<~TS
+        /* eslint-disable react/no-unescaped-entities */
+        /* eslint-disable @typescript-eslint/naming-convention */
+        #{import_block}
+
+        type ContentComponent = (props: {styles: ThemeStyles}) => ReactNode;
+
+        type HelpContent = {
+            /** The content to display for this route */
+            content: ContentComponent;
+
+            /** Any children routes that this route has */
+            children?: Record<string, HelpContent>;
+
+            /** Whether this route is an exact match or displays parent content */
+            isExact?: boolean;
+        };
+
+        const helpContentMap: HelpContent = #{help_content_string}
+
+        export default helpContentMap;
+        export type {ContentComponent};
+      TS
+            
+      File.write(output_file, ts_output)
+
+      puts "✅ Successfully generated helpContent.tsx"
+    end
+
+    def self.generate_help_content_tree()
+      tree = {}
+    
+      @help_mapping.each do |route, node|
+        parts = route.sub(/^ref\//, '').sub(/\.md$/, '').split('/')
+        current = tree
+    
+        parts.each_with_index do |part, i|
+          is_dynamic = part.start_with?(':') || part.match?(/^\[.*\]$/)
+          part_key = is_dynamic ? part : part.to_sym
+    
+          current[:children] ||= {}
+          current[:children][part_key] ||= {}
+    
+          if i == parts.length - 1
+            jsx_content = html_node_to_RN(node, 1).rstrip
+    
+            current[:children][part_key][:content] = <<~TS.chomp
+              ({styles}: {styles: ThemeStyles}) => (
+              #{jsx_content}
+              )
+            TS
+          end
+    
+          current = current[:children][part_key]
+        end
       end
+    
+      tree[:content] = <<~JSX
+        () => null
+      JSX
+      tree
+    end
+    
+    def self.html_node_to_RN(node, indent_level = 0)
+      indent = '  ' * indent_level
```

Comment:
Should if add a check `return "" if node.nil?` ?

Reply:
Hm... Why would it be `nil`?